jib-core/src/main/java/com/google/cloud/tools/jib/image/json/V22ManifestListTemplate.java

```diff
@@ -0,0 +1,171 @@
+ *
+ * <p>Example manifest JSON:
+ *
+ * <pre>{@code
+ * {
+ *   "schemaVersion": 2,
+ *   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
+ *   "manifests": [
+ *     {
+ *       "mediaType": "application/vnd.docker.image.manifest.v2+json",
+ *       "size": 7143,
+ *       "digest": "sha256:e692418e4cbaf90ca69d05a66403747baa33ee08806650b51fab815ad7fc331f",
+ *       "platform": {
+ *         "architecture": "ppc64le",
+ *         "os": "linux",
+ *       }
+ *     },
+ *     {
+ *       "mediaType": "application/vnd.docker.image.manifest.v2+json",
+ *       "size": 7682,
+ *       "digest": "sha256:5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270",
+ *       "platform": {
+ *         "architecture": "amd64",
+ *         "os": "linux",
+ *         "features": [
+ *           "sse4"
+ *         ]
+ *       }
+ *     }
+ *   ]
+ * }
+ * }</pre>
+ *
+ * @see <a href="https://docs.docker.com/registry/spec/manifest-v2-2/#manifest-list">Image Manifest
+ *     Version 2, Schema 2: Manifest List</a>
+ */
+public class V22ManifestListTemplate implements ManifestTemplate {
+
+  public static final String MANIFEST_MEDIA_TYPE =
+      "application/vnd.docker.distribution.manifest.list.v2+json";
+  private static final int SCHEMA_VERSION = 2;
+
+  @Override
+  public int getSchemaVersion() {
+    return SCHEMA_VERSION;
+  }
+
+  private List<ManifestDescriptorTemplate> manifests = new ArrayList<>();
+
+  @VisibleForTesting
+  List<ManifestDescriptorTemplate> getManifests() {
+    return manifests;
+  }
+
+  public List<String> getDigestForPlatform(String architecture, String os) {
+    return manifests
+        .stream()
+        .filter(
+            manifest -> {
+              if (manifest.platform == null) {
+                return false;
+              }
+              if (!os.equals(manifest.platform.os)) {
+                return false;
+              }
+              if (!architecture.equals(manifest.platform.architecture)) {
+                return false;
+              }
+              return true;
+            })
+        .map(ManifestDescriptorTemplate::getDigest)
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Template for inner JSON object representing content descriptor for a layer or container
+   * configuration.
+   *
+   * @see <a href="https://github.com/opencontainers/image-spec/blob/master/descriptor.md">OCI
+   *     Content Descriptors</a>
+   */
+  static class ManifestDescriptorTemplate implements JsonTemplate {
+
+    static class Platform implements JsonTemplate {
+      @Nullable private String architecture;
+      @Nullable private String os;
+      // ignored properties
+      @Nullable
+      @JsonProperty("os.version")
+      private String osVersion;
+
+      @Nullable
+      @JsonProperty("os.features")
+      private List<String> osFeatures;
+
+      @Nullable private String variant;
+      @Nullable private List<String> features;
```

Comment:
Any rule for this type of field, given that you did `private List<ManifestDescriptorTemplate> manifests = new ArrayList<>()` above?

Reply:
I think features is technically nullable in that the json doesn't **need** to include it. However manifest **must** be defined in the json.

Reply:
I changed all this to be consistent.